### PR TITLE
fix Popover controlled mode during first render

### PR DESCRIPTION
**Pull request by mayank99**
## Changes

Ported from https://github.com/iTwin/iTwinUI/pull/1265#issuecomment-1545755572 for release in v2. It could fix some of the issues with our Popover.

## Testing

Fixes the weird edge cases that I was facing in #1265, and doesn't cause any existing tests to break, which is good.

## Docs

N/A

### CODE REVIEW

**Comment by mayank99** [comment]
This could maybe help with https://github.com/iTwin/appui/issues/271 (unsure).

**Review by LostABike — APPROVED** [review]
This is the same change in #1265 , wouldn't that just override it?

**Comment by mayank99** [comment]
> This is the same change in #1265 , wouldn't that just override it?

This PR is meant to be released in the next v2 release (#1250), whereas #1265 is a breaking change that won't be released until v3 (a few months from now).

File: packages/itwinui-react/src/core/utils/components/Popover.tsx [hunk 1]
```diff
@@ -156,3 +161,9 @@ export const hideOnEscOrTab = {
 };
 
 export default Popover;
+
+const useIsDomAvailable = () => {
```
Comment (gretanausedaite) [inline packages/itwinui-react/src/core/utils/components/Popover.tsx:165]: Does it belong in hooks folder?

**Review by gretanausedaite — APPROVED** [review]
LGTM
